src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -681,42 +711,56 @@ BuildShardPlacementList(ShardInterval *shardInterval)
 {
 	int64 shardId = shardInterval->shardId;
 	List *shardPlacementList = NIL;
-	Relation pgShardPlacement = NULL;
+	Relation pgPlacement = NULL;
 	SysScanDesc scanDescriptor = NULL;
 	ScanKeyData scanKey[1];
 	int scanKeyCount = 1;
 	bool indexOK = true;
 	HeapTuple heapTuple = NULL;
 
-	pgShardPlacement = heap_open(DistShardPlacementRelationId(), AccessShareLock);
+	pgPlacement = heap_open(DistPlacementRelationId(), AccessShareLock);
 
-	ScanKeyInit(&scanKey[0], Anum_pg_dist_shard_placement_shardid,
+	ScanKeyInit(&scanKey[0], Anum_pg_dist_placement_shardid,
 				BTEqualStrategyNumber, F_INT8EQ, Int64GetDatum(shardId));
 
-	scanDescriptor = systable_beginscan(pgShardPlacement,
-										DistShardPlacementShardidIndexId(), indexOK,
+	scanDescriptor = systable_beginscan(pgPlacement,
+										DistPlacementShardidIndexId(), indexOK,
 										NULL, scanKeyCount, scanKey);
 
 	heapTuple = systable_getnext(scanDescriptor);
 	while (HeapTupleIsValid(heapTuple))
 	{
-		TupleDesc tupleDescriptor = RelationGetDescr(pgShardPlacement);
+		TupleDesc tupleDescriptor = RelationGetDescr(pgPlacement);
 
 		ShardPlacement *placement = TupleToShardPlacement(tupleDescriptor, heapTuple);
+
+		uint32 groupId = placement->groupId;
+		WorkerNode *worker = NodeForGroup(groupId);
+
+		if (worker == NULL)
+		{
+			ereport(ERROR, (errmsg("the metadata is inconsistent"),
```

Comment:
I considered two options:
- Don't return any ShardPlacements without a matching node, the other functions don't even see them
- Leave an empty nodeName/nodePort. 

The first one seemed nice but difficult to revert if we later had a function which wanted inactive shard placements.

The second one seemed bad for the same reasons removing nodeName/nodePort from ShardPlacement seemed bad, it potentially changes a lot of call sites!

I'm looking into it now, coming up with a list of functions which need to be changed.